web/crux-ui/src/components/notifications/edit-notification-card.tsx

```diff
@@ -0,0 +1,167 @@
+import { DyoButton } from '@app/elements/dyo-button'
+import { DyoCard } from '@app/elements/dyo-card'
+import DyoChips from '@app/elements/dyo-chips'
+import { DyoHeading } from '@app/elements/dyo-heading'
+import { DyoInput } from '@app/elements/dyo-input'
+import { DyoLabel } from '@app/elements/dyo-label'
+import { DyoSwitch } from '@app/elements/dyo-switch'
+import { defaultApiErrorHandler } from '@app/errors'
+import {
+  CreateNotification,
+  NotificationDetails,
+  NotificationItem,
+  NotificationType,
+  NOTIFICATION_TYPE_VALUES,
+  UpdateNotification,
+} from '@app/models'
+import { notificationApiHookUrl, notificationApiUrl } from '@app/routes'
+import { sendForm } from '@app/utils'
+import { notificationSchema } from '@app/validation'
+import { useFormik } from 'formik'
+import useTranslation from 'next-translate/useTranslation'
+import { MutableRefObject, useEffect, useState } from 'react'
+import toast from 'react-hot-toast'
+
+interface EditNotificationCardProps {
+  notification?: NotificationDetails
+  submitRef: MutableRefObject<() => Promise<any>>
+  onSubmitted: (notifcation: NotificationItem) => void
+  className?: string
+}
+
+const EditNotificationCard = (props: EditNotificationCardProps) => {
+  const { t } = useTranslation('notifications')
+
+  const [notification, setNotification] = useState<NotificationDetails>(
+    props.notification ?? {
+      id: null,
+      active: true,
+      name: '',
+      type: 'discord',
+      url: '',
+      createdBy: '',
+    },
+  )
+
+  const isEditMode = !!notification.id
+
+  const handleApiError = defaultApiErrorHandler(t)
+
+  const formik = useFormik({
+    initialValues: {
+      ...notification,
+    },
+    validationSchema: notificationSchema,
+    onSubmit: async (values, { setSubmitting, setFieldError }) => {
+      setSubmitting(true)
+
+      const request: CreateNotification | UpdateNotification = {
+        ...values,
+      }
+
+      const response = await (isEditMode
+        ? sendForm('PUT', notificationApiUrl(notification.id), request as UpdateNotification)
+        : sendForm('POST', notificationApiUrl(), request as CreateNotification))
+
+      if (response.ok) {
+        const result = response.status == 200 ? ((await response.json()) as NotificationDetails) : { ...values }
+        setNotification(result)
+        props.onSubmitted(result as NotificationItem)
+      } else {
+        handleApiError(response, setFieldError)
+      }
+
+      setSubmitting(false)
+    },
+  })
+
+  if (props.submitRef) {
+    props.submitRef.current = formik.submitForm
+  }
+
+  const [testDisabled, setTestDisabled] = useState<boolean>(true)
```

Comment:
move this up to the other usestates pls.
Also the <boolean> is unnecessary, cause ts can infer it from the initial value